build.gradle

```diff
@@ -21,3 +21,6 @@ task clean(type: Delete) {
     delete rootProject.buildDir
 }
 
+task buildTestConfig(type:Exec) {
+  commandLine 'bin/build_test_config.sh'
+}
```

Comment:
Potentially -- towards benefit of the developer use case -- you could declare this as a dependency for the instrumented tests task. Maybe something like this would work?
```
connectedAndroidTest.dependsOn buildTestConfig
```

Reply:
(I had no idea what I was doing for this part)

Reply:
I always get sandbagged on this, whenever I have to add task dependencies. Seems like it's never as straightforward as it ought to be.